### PR TITLE
Support HyperVGeneration attribute for shared_image

### LINKED ISSUE #6316: Attribute missing from resource azurerm_shared_image
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The attribute hyper-V-generation is not supported by resource azurerm_shared_image

### New or Affected Resource(s)

* azurerm_shared_image

### Potential Terraform Configuration

```hcl
resource "azurerm_shared_image" "c8v2" {
  name                = "${local.namebase}-c8v2"
  description         = "Centos8 image with Folksam customizations. Derived from OpenLogic's image."
  gallery_name        = azurerm_shared_image_gallery.linux.name
  resource_group_name = azurerm_resource_group.this.name
  location            = azurerm_resource_group.this.location
  os_type             = "Linux"
  hyper-v-generation = "V2"
  identifier {
    publisher = "Folksam"
    offer     = "CentOS"
    sku       = "c8v2"
  }
}
```

### References

* https://docs.microsoft.com/en-us/cli/azure/sig/image-definition?view=azure-cli-latest#az-sig-image-definition-create

### PR DESCRIPTION
**Pull request by manicminer**
- Adds attribute to azurerm_shared_image resource and data.azurerm_shared_image
- Workaround for consistency issue when deleting shared images (by @neil-yechenwei )

References #6495   
Fixes #6316

### CODE REVIEW

File: website/docs/d/shared_image.html.markdown [hunk 1]
```diff
@@ -47,6 +47,8 @@ The following attributes are exported:
 
 * `os_type` - The type of Operating System present in this Shared Image.
 
+* `hyper_v_generation` - The hypervisor generation of the virtual machine.
```
Comment (tombuildsstuff) [inline website/docs/d/shared_image.html.markdown:50]:
```suggestion
* `hyper_v_generation` - The generation of the HyperV the Virtual Machine the image is based on was created from.
```

File: azurerm/internal/services/compute/tests/resource_arm_shared_image_test.go [hunk 2]
```diff
@@ -138,7 +160,12 @@ func testCheckAzureRMSharedImageExists(resourceName string) resource.TestCheckFu
 	}
 }
 
-func testAccAzureRMSharedImage_basic(data acceptance.TestData) string {
+func testAccAzureRMSharedImage_basic(data acceptance.TestData, hyperVGen string) string {
+	hyperVGenAttr := ""
+	if hyperVGen != "" {
+		hyperVGenAttr = fmt.Sprintf(`hyper_v_generation = "%s"`, hyperVGen)
+	}
```
[the remark this answers is not in the document]
Comment (manicminer) [reply azurerm/internal/services/compute/tests/resource_arm_shared_image_test.go:167]: Thanks, done!

File: website/docs/r/shared_image.html.markdown [hunk 3]
```diff
@@ -68,6 +68,8 @@ The following arguments are supported:
 
 * `eula` - (Optional) The End User Licence Agreement for the Shared Image.
 
+* `hyper_v_generation` - (Optional) The HyperVGenerationType of the Virtual Machine created from the image. Possible values are `V1` and `V2`. Defaults to `V1`. Changing this forces a new resource to be created.
```
[the remark this answers is not in the document]
Comment (manicminer) [reply website/docs/r/shared_image.html.markdown]: Thanks, done!

File: website/docs/d/shared_image.html.markdown (hunk 1 above)
Comment (manicminer) [reply website/docs/d/shared_image.html.markdown]: Thanks, done!

**Pull request merged**